File: src/cn/edustar/jitar/action/BaseChannelManage.java
package cn.edustar.jitar.action;

import cn.edustar.jitar.pojos.AccessControl;
import cn.edustar.jitar.pojos.Channel;
import cn.edustar.jitar.service.AccessControlService;
import cn.edustar.jitar.service.ChannelPageService;
import cn.edustar.jitar.util.CommonUtil;

/**
 * 自定义频道后台管理的新基类
 * 
 * @author baimindong
 *
 */
public abstract class BaseChannelManage extends ManageBaseAction  {
	
	private static final long serialVersionUID = 81039722590779603L;

	protected ChannelPageService channelPageService;
	protected AccessControlService accessControlService;
	
    public boolean isSystemAdmin(){
    	if (!isUserLogined()){return false;}
        return accessControlService.isSystemAdmin(getLoginUser());
    }
    
    public boolean isChannelSystemAdmin(Channel channel){
    	if (!isUserLogined()){return false;}
        if (accessControlService.isSystemAdmin(getLoginUser())){return true;}
        return accessControlService.checkUserAccessControlIsExists(getLoginUser().getUserId(), AccessControl.OBJECTTYPE_CHANNELSYSTEMADMIN, channel.getChannelId());
    }
    
    public boolean isChannelUserAdmin(Channel channel){
    	if (!isUserLogined()){return false;}
    	if (accessControlService.isSystemAdmin(getLoginUser())){return true;}
        return accessControlService.checkUserAccessControlIsExists(getLoginUser().getUserId(), AccessControl.OBJECTTYPE_CHANNELUSERADMIN, channel.getChannelId());
    }
    
    public boolean isChannelContentAdmin(Channel channel){
    	if (!isUserLogined()){return false;}
    	if (accessControlService.isSystemAdmin(getLoginUser())){return true;}
        return accessControlService.checkUserAccessControlIsExists(getLoginUser().getUserId(), AccessControl.OBJECTTYPE_CHANNELCONTENTADMIN, channel.getChannelId());
    }
    
    public String GetAdminType(Channel channel){
        String AdminType = "|";
        if(accessControlService.isSystemAdmin(getLoginUser())){
            AdminType = AdminType + "SystemSuperAdmin|";
        }
        if(accessControlService.isSystemUserAdmin(getLoginUser())){
            AdminType = AdminType + "SystemUserAdmin|";
        }
        if(accessControlService.isSystemContentAdmin(getLoginUser())){
            AdminType = AdminType + "SystemContentAdmin|";
        }
        if(accessControlService.getAccessControlByUserAndObject(getLoginUser().getUserId(), AccessControl.OBJECTTYPE_CHANNELSYSTEMADMIN, channel.getChannelId()) != null){
            AdminType = AdminType + "ChannelSystemAdmin|";
        }
        if(accessControlService.getAccessControlByUserAndObject(getLoginUser().getUserId(), AccessControl.OBJECTTYPE_CHANNELUSERADMIN, channel.getChannelId()) != null){
            AdminType = AdminType + "ChannelUserAdmin|";
        }
        if(accessControlService.getAccessControlByUserAndObject(getLoginUser().getUserId(), AccessControl.OBJECTTYPE_CHANNELCONTENTADMIN, channel.getChannelId()) != null){
            AdminType = AdminType + "ChannelContentAdmin|";
        }
        if(AdminType.equals("|")){AdminType = "";}
        return AdminType;
    }
    
    public String getSiteUrl(){
        return CommonUtil.getSiteUrl(request);
    }
        		
	public ChannelPageService getChannelPageService() {
		return channelPageService;
	}

	public void setChannelPageService(ChannelPageService channelPageService) {
		this.channelPageService = channelPageService;
	}

	public AccessControlService getAccessControlService() {
		return accessControlService;
	}

	public void setAccessControlService(AccessControlService accessControlService) {
		this.accessControlService = accessControlService;
	}
}

File: WebContent/WEB-INF/jython/base_channel_manage.py
# encoding=utf-8
from cn.edustar.jitar.jython import JythonBaseAction
from cn.edustar.jitar.util import ParamUtil, CommonUtil
from cn.edustar.jitar.pojos import AccessControl

# 后台管理的新基类
class BaseChannelManage(JythonBaseAction):
    ERROR = "/WEB-INF/ftl/Error.ftl"
    ACCESS_DENIED = "/WEB-INF/ftl/Error.ftl"
    ACCESS_ERROR = "/WEB-INF/ftl/mengv1/access_error.ftl"
    SUCCESS = "/WEB-INF/ftl/success.ftl"
    LOGIN = "login.jsp"
    
    def __init__(self):
        self.channelPageService = __spring__.getBean("channelPageService")
        self.params = ParamUtil(request)
        if self.loginUser == None:
            response.sendRedirect(self.getSiteUrl() + self.LOGIN + "?redUrl=" + self.getSiteUrl() + "manage/channel/index.py")    
        
    def isSystemAdmin(self):
        if self.loginUser == None:return False
        accessControlService = __spring__.getBean("accessControlService")
        if accessControlService.isSystemAdmin(self.loginUser) == True:
            return True
        return accessControlService.isSystemAdmin(self.loginUser)
    
    def isChannelSystemAdmin(self, channel):
        accessControlService = __spring__.getBean("accessControlService")
        return accessControlService.checkUserAccessControlIsExists(self.loginUser.userId, AccessControl.OBJECTTYPE_CHANNELSYSTEMADMIN, channel.channelId)
    
    def isChannelUserAdmin(self, channel):
        accessControlService = __spring__.getBean("accessControlService")
        return accessControlService.checkUserAccessControlIsExists(self.loginUser.userId, AccessControl.OBJECTTYPE_CHANNELUSERADMIN, channel.channelId)
    
    def isChannelContentAdmin(self, channel):
        accessControlService = __spring__.getBean("accessControlService")
        return accessControlService.checkUserAccessControlIsExists(self.loginUser.userId, AccessControl.OBJECTTYPE_CHANNELCONTENTADMIN, channel.channelId)
    
    def GetAdminType(self, channel):
        AdminType = "|"
        accessControlService = __spring__.getBean("accessControlService")
        if accessControlService.isSystemAdmin(self.loginUser):
            AdminType = AdminType + "SystemSuperAdmin|"
        if accessControlService.isSystemUserAdmin(self.loginUser):
            AdminType = AdminType + "SystemUserAdmin|"
        if accessControlService.isSystemContentAdmin(self.loginUser):
            AdminType = AdminType + "SystemContentAdmin|"
        if accessControlService.getAccessControlByUserAndObject(self.loginUser.userId, AccessControl.OBJECTTYPE_CHANNELSYSTEMADMIN, channel.channelId) != None:
            AdminType = AdminType + "ChannelSystemAdmin|"
        if accessControlService.getAccessControlByUserAndObject(self.loginUser.userId, AccessControl.OBJECTTYPE_CHANNELUSERADMIN, channel.channelId) != None:
            AdminType = AdminType + "ChannelUserAdmin|"
        if accessControlService.getAccessControlByUserAndObject(self.loginUser.userId, AccessControl.OBJECTTYPE_CHANNELCONTENTADMIN, channel.channelId) != None:
            AdminType = AdminType + "ChannelContentAdmin|"
        if AdminType == "|":AdminType = ""
        return AdminType
    def getSiteUrl(self):
        return CommonUtil.getSiteUrl(request)
